c/cert/src/rules/ERR33-C/DetectAndHandleStandardLibraryErrors.ql

```diff
@@ -0,0 +1,207 @@
+/**
+ * @id c/cert/detect-and-handle-standard-library-errors
+ * @name ERR33-C: Detect and handle standard library errors
+ * @description Detect and handle standard library errors.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/err33-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import semmle.code.cpp.commons.NULL
+import codingstandards.cpp.ReadErrorsAndEOF
+import semmle.code.cpp.dataflow.DataFlow
+
+/**
+ * Classifies error returning function calls based on the
+ * type and value of the required checked
+ */
+class ExpectedErrReturn extends FunctionCall {
```

Comment:
recently on my Preprocessor6 package Remco [gave me some feedback that abstract classes](https://github.com/github/codeql-coding-standards/pull/99/files#diff-026dd0360dc3233a8e85504b5c38c638d74caa50800b07a43335f8eaf399fc31R8) were a neat way to do some case handling, I think considering either using abstract classes or at least more types for the specific cases covered here might make this more readable